File: 백준/Silver/10845. 큐/큐.py
from collections import deque
import sys
input = sys.stdin.readline

def command(x):
    if x[0] == 'push':
        arr.append(x[1])
    elif x[0] == 'pop':
        if len(arr) == 0:
            print(-1)
        else:
            print(arr.popleft())
    elif x[0] == 'size':
        print(len(arr))
    elif x[0] == 'empty':
        if len(arr) == 0:
            print(1)
        else:
            print(0)
    elif x[0] == 'front':
        if len(arr) == 0:
            print(-1)
        else:
            print(arr[0])
    elif x[0] == 'back':
        if len(arr) == 0:
            print(-1)
        else:
            print(arr[-1])
            
n = int(input())
arr = deque([])
for _ in range(n):
    command(list(input().split()))

File: 백준/Silver/10845. 큐/큐.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayDeque;
import java.util.Deque;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		String input;
		Deque<Integer> q = new ArrayDeque<>();
		
		for (int i = 0; i < n; i++) {
			input = br.readLine();
			switch (input) {
			case ("pop"):
				System.out.println(q.isEmpty() ? -1 : q.pollFirst());
				break;
			case ("size"):
				System.out.println(q.size());
				break;
			case ("empty"):
				System.out.println(q.isEmpty() ? 1 : 0);
				break;
			case ("front"):
				System.out.println(q.isEmpty() ? -1 : q.getFirst());
				break;
			case ("back"):
				System.out.println(q.isEmpty() ? -1 : q.getLast());
				break;
			default:
				q.addLast(Integer.parseInt(input.split(" ")[1]));
			}
		}

	}
}
